nanoprintf.h

```diff
@@ -894,9 +896,9 @@ int npf_vpprintf(npf_putc pc, void *pc_ctx, char const *format, va_list args) {
         } else
 #endif
         {
-          unsigned const base = (fs.conv_spec == NPF_FMT_SPEC_CONV_OCTAL) ?
+          uint_fast8_t const base = (fs.conv_spec == NPF_FMT_SPEC_CONV_OCTAL) ?
```

Comment:
In previous iterations / compilers, I've found that using non-register-width types can incur an extra instruction penalty for masking off the upper unused bits (if any). Using native register sizes (which I wish C had a type for) let the compiler emit smaller code because it knew that these expression used the whole register width. Does this change cause an interesting change with respect to the generated assembly?

Reply:
That would be the case with `uint8_t` and `uint_least8_t`, but not the `uint_fast8_t`. Here is an [example](https://stackoverflow.com/a/59593614) of how the _fast_ `stdint.h` types are defined on different platforms. These types have been introduced exactly for this purpose - to be portable and optimal at the same time. Essentially they provide the functionality actually necessary in real life, contrary to the C's base types.

On 32-bit ARM the `uint_fast8_t` is actually the same 32-bit unsigned integer and doesn't change anything at all. Because the the `int` is not allowed to be smaller than 16-bit, the biggest impact is on CPUs smaller than that. The best popular example is the 8-bit AVR, but for the 16-bit MSP430 and RL78 the code also tends to be smaller. On AVR I basically can observe that for every place, where the 16-bit `int/unsigned/enum` is accessed, the binary size increases by additional 2, 4 or more bytes compared to 8-bit data types.